os/board/common/crashdump.c

```diff
@@ -105,28 +106,58 @@ static int ramdump_via_uart(void)
 	/* Send number of memory regions to HOST */
 	up_lowputc(CONFIG_MM_REGIONS);
 
+	/* Send number of memory regions to HOST */
+	up_lowputc(CONFIG_KMM_REGIONS);
+
+	/* Send number of memory regions to HOST */
+	up_lowputc((CONFIG_MM_REGIONS + CONFIG_KMM_REGIONS));
+
 	/* Send memory region address, size & heap index to HOST */
-	for (x = 0; x < CONFIG_MM_REGIONS; x++) {
+	for (x = 0; x < (CONFIG_MM_REGIONS + CONFIG_KMM_REGIONS); x++) {
 
-		/* Send RAM address */
-		ptr = (uint8_t *)&regionx_start[x];
-		for (i = 0; i < sizeof(regionx_start[x]); i++) {
-			up_lowputc((uint8_t)*ptr);
-			ptr++;
-		}
+		if (x < CONFIG_MM_REGIONS) {
 
-		/* Send RAM size */
-		ptr = (uint8_t *)&regionx_size[x];
-		for (i = 0; i < sizeof(regionx_size[x]); i++) {
-			up_lowputc((uint8_t)*ptr);
-			ptr++;
+			/* Send RAM address */
+			ptr = (uint8_t *)&regionx_start[x];
+			for (i = 0; i < sizeof(regionx_start[x]); i++) {
+				up_lowputc((uint8_t)*ptr);
+				ptr++;
+			}
+
+			/* Send RAM size */
+			ptr = (uint8_t *)&regionx_size[x];
+			for (i = 0; i < sizeof(regionx_size[x]); i++) {
+				up_lowputc((uint8_t)*ptr);
+				ptr++;
+			}
+
+			/* Send Heap Index */
+			up_lowputc(regionx_heap_idx[x]);
 		}
 
-		/* Send Heap Index */
-		up_lowputc(regionx_heap_idx[x]);
+		else if (x >= CONFIG_MM_REGIONS) {
+			y =  x - CONFIG_MM_REGIONS;
+
+			/* Send Kernel region address */
+			ptr = (uint8_t *)&kregionx_start[y];
```

Comment:
kregionx_start is valid with CONFIG_MM_KERNEL_HEAP enabled. But here is no conditional for that. Could you confirm this is working on CONFIG_MM_KERNEL_HEAP disabled like ARTIK?

Reply:
You're right, CONFIG_MM_KERNEL_HEAP dependency needs to be added s build breaks with protected kernel heap config disabled. I will add it